sass-style-guide.md

```diff
@@ -59,87 +42,111 @@ You should order your rules from generic to specific. Example:
 
 // Abstract Component Imports
 //---------------------------
-@import abstract-components/section-box
+@import abstract-components/boundary-box
+@import abstract-components/gradient-overlay
 
 // Generic Component Imports
 //--------------------------
 @import generic-components/button
-@import generic-components/skip-links
+@import generic-components/data-table
 
 // Specific Component Imports
 //---------------------------
-@import specific-components/page/page-footer
-@import specific-components/page/page-header
+@import specific-components/company-search
+@import specific-components/page-header
+```
+
+## Style-Definitionen
+
+Wann wird wo was definiert? Wir gehen inhaltsgetrieben vor, d.h. wir schreiben unseren Sass-Code in der Reihenfolge, wie die Elemente auf der gerenderten Seite erscheinen werden:
+
+1. **von außen nach innen**: ZASAF folgend beginnt man in der Regel mit dem Page-Wrapper-Element.
+2. **von oben nach unten**: Danach unterteilt man sich die Seite in (horizontale) Sektionen und beginnt mit der ersten (page-header -> page-main -> page-footer).
+3. **von links nach rechts**: In einer Sektion fängt man ganz links mit dem ersten Element an (Page-Header: Logo links -> Navigation in der Mitte -> Suche rechts).
+
+### Notationsreihenfolge innerhalb eines Elements
+
+Verschachtelungen innerhalb eines Elements beginnen mit Pseudoklassen, gefolgt von Pseudoelementen, gefolgt von Modifiern. Auf Kindelemente, sowohl direkte als auch tiefer im DOM verschachtelte, sollte weitestgehend verzichtet werden. Hierfür wäre es besser einen neuen Selektorblock zu erstellen (siehe Coding Konventionen unten).
+
+``` sass
+.main-navigation--item
+  position: relative
+  &:hover
+    border-color: $color-border-loud
+  &::before
+    content: '*'
+  &.highlight
+    bacground-color: $color-background-loud
+
+.main-navigation--item
+  display: block
+  padding: 5px 0
+  .highlight > &
```

Comment:
Das ist eine Grundregel von Modifiern, s. https://github.com/zweitag/html-css-guidelines/blob/master/README.md#modifier:

> steht nie alleine, sondern ist nur in Verbindung mit der Komponente eindeutig

Ein Modifier an sich ist eben nicht eindeutig. Es können viele Dinge `.highlighted` sein. Nun ist die `.main-navigation` immer sehr hoch in der Seitenhierarchie angesiedelt. Aber nimm eine beliebige Liste, die gehighlightete Einträge enthalten kann, die wiederum Inhalt enthalten kann, der gehighlighted sein kann:

``` haml
%ul.articles-list
  %li.articles-list--item.highlighted
    %h3 Hitlers geheimes Silberbesteck gefunden
    %ul.tag-cloud
      %li.tag-cloud--item.highlighted
        %a.tag-cloud--link Hitler
      %li.tag-cloud--item
        %a.tag-cloud--link Sommerflautenunfug
  %li.articles-list--item.highlighted
    %h3 Große Dürre: Werden wir alle sterben?
    %ul.tag-cloud
      %li.tag-cloud--item
        %a.tag-cloud--link Wetter
      %li.tag-cloud--item
        %a.tag-cloud--link Sommerflautenunfug
```

Wenn du jetzt schreibst…:

``` sass
.tag-cloud--link
  &.highlighted
    background-color: yellow
```

… dann sind in obigem Beispiel alle Tag-Cloud-Links gelb hinterlegt, obwohl es nur der mit "Hitler" sein sollte.

Daher die Regel: Modifier dürfen nie alleine stehen. Alternativ müsste man es m.M.n. so machen wie BEM und `%a.tag-cloud--link.tag-cloud--link__highlighted` schreiben oder sowas.